### PR TITLE
Correct the implementation of the buzzer feature

### PR DESCRIPTION
**Pull request by jpptm**
Generalise the access to the buzzer message by adding a frequency field so other modules can use this message as required.

### CODE REVIEW

File: module/platform/OpenCR/HardwareIO/src/hardwareio/process_data.cpp [hunk 1]
```diff
@@ -133,7 +133,9 @@ namespace module::platform::OpenCR {
 
         for (const auto& servo : servo_states) {
             if (servo.temperature > cfg.max_tol_temp) {
-                emit(std::make_unique<Buzzer>());
+                auto buzzer_msg = std::make_unique<Buzzer>();
+                buzzer_msg -> buzzer_frequency = cfg.buzzer_freq;
+                emit(buzzer_msg);
```
Comment (ysims) [inline module/platform/OpenCR/HardwareIO/src/hardwareio/process_data.cpp:138]:
```suggestion
                emit(std::make_unique<Buzzer>(cfg.buzzer_freq));
```

File: shared/message/output/Buzzer.proto [hunk 2]
```diff
@@ -2,4 +2,6 @@ syntax = "proto3";
 
 package message.output;
 
-message Buzzer {}
+message Buzzer {
+    int32 buzzer_frequency = 1;
+}
```
Comment (ysims) [inline shared/message/output/Buzzer.proto:7]: Put a comment on the buzzer message. It isn't clear what is buzzing to someone who isn't that familiar with this kind of stuff.

Comment (jpptm) [reply shared/message/output/Buzzer.proto:11]: Fixed

**Comment by ysims** [comment]
This PR is to merge into robocup2023. It doesn't make sense to add to this anymore, you may as well make a PR to main with the buzzer functionality.

**Comment by jpptm** [comment]
@ysims just change the merge to main?

**Comment by jpptm** [comment]
To merge properly tested code into main instead

**Comment by ysims** [comment]
Branch should be deleted if not needed anymore

**Pull request closed without merging**